homeassistant/components/netatmo/climate.py

```diff
@@ -85,63 +91,67 @@
+        vol.Required(ATTR_HOME_NAME): cv.string,
+    }
+)
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the NetAtmo Thermostat."""
-    homes_conf = config.get(CONF_HOMES)
 
-    auth = hass.data[DATA_NETATMO_AUTH]
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Netatmo energy platform."""
+    auth = hass.data[DOMAIN][AUTH]
 
     home_data = HomeData(auth)
-    try:
-        home_data.setup()
-    except pyatmo.NoDevice:
-        return
-
-    home_ids = []
-    rooms = {}
-    if homes_conf is not None:
-        for home_conf in homes_conf:
-            home = home_conf[CONF_NAME]
-            home_id = home_data.homedata.gethomeId(home)
-            if home_conf[CONF_ROOMS] != []:
-                rooms[home_id] = home_conf[CONF_ROOMS]
-            home_ids.append(home_id)
-    else:
-        home_ids = home_data.get_home_ids()
 
-    devices = []
-    for home_id in home_ids:
-        _LOGGER.debug("Setting up %s ...", home_id)
+    def get_devices():
+        """Retrieve Netatmo devices."""
+        devices = []
         try:
-            room_data = ThermostatData(auth, home_id)
+            home_data.setup()
         except pyatmo.NoDevice:
-            continue
-        for room_id in room_data.get_room_ids():
-            room_name = room_data.homedata.rooms[home_id][room_id]["name"]
-            _LOGGER.debug("Setting up %s (%s) ...", room_name, room_id)
-            if home_id in rooms and room_name not in rooms[home_id]:
-                _LOGGER.debug("Excluding %s ...", room_name)
+            return
+        home_ids = home_data.get_all_home_ids()
+
+        for home_id in home_ids:
+            _LOGGER.debug("Setting up home %s ...", home_id)
+            try:
+                room_data = ThermostatData(auth, home_id)
+            except pyatmo.NoDevice:
                 continue
-            _LOGGER.debug("Adding devices for room %s (%s) ...", room_name, room_id)
-            devices.append(NetatmoThermostat(room_data, room_id))
-    add_entities(devices, True)
+            for room_id in room_data.get_room_ids():
+                room_name = room_data.homedata.rooms[home_id][room_id]["name"]
+                _LOGGER.debug("Setting up room %s (%s) ...", room_name, room_id)
+                devices.append(NetatmoThermostat(room_data, room_id))
+        return devices
+
+    async_add_entities(await hass.async_add_executor_job(get_devices), True)
+
+    def _service_setschedule(service):
+        """Service to change current home schedule."""
+        home_name = service.data.get(ATTR_HOME_NAME)
+        schedule_name = service.data.get(ATTR_SCHEDULE_NAME)
+        home_data.homedata.switchHomeSchedule(schedule=schedule_name, home=home_name)
+        _LOGGER.info("Set home (%s) schedule to %s", home_name, schedule_name)
+
+    if home_data.homedata is not None:
+        hass.services.async_register(
```

Comment:
Register service in the component (`__init__.py`) during async_setup_entry.

Reply:
Ok, but why? And what if no climate devices are detected?